app/src/main/java/in/testpress/testpress/ui/OrderConfirmActivity.java

```diff
@@ -0,0 +1,293 @@
+import android.os.Bundle;
+import android.text.Editable;
+import android.text.TextWatcher;
+import android.view.KeyEvent;
+import android.view.MenuItem;
+import android.view.View;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.ProgressBar;
+import android.widget.RelativeLayout;
+import android.widget.TextView;
+
+import com.afollestad.materialdialogs.MaterialDialog;
+import com.payu.india.Model.PaymentParams;
+import com.payu.india.Model.PayuConfig;
+import com.payu.india.Model.PayuHashes;
+import com.payu.india.Payu.PayuConstants;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.inject.Inject;
+
+import butterknife.ButterKnife;
+import butterknife.InjectView;
+import butterknife.OnClick;
+import in.testpress.testpress.Injector;
+import in.testpress.testpress.R;
+import in.testpress.testpress.R.id;
+import in.testpress.testpress.TestpressServiceProvider;
+import in.testpress.testpress.authenticator.LogoutService;
+import in.testpress.testpress.core.Constants;
+import in.testpress.testpress.models.Order;
+import in.testpress.testpress.models.OrderItem;
+import in.testpress.testpress.models.ProductDetails;
+import in.testpress.testpress.ui.paymentGateway.PaymentModeActivity;
+import in.testpress.testpress.util.InternetConnectivityChecker;
+import in.testpress.testpress.util.SafeAsyncTask;
+
+import static android.view.inputmethod.EditorInfo.IME_ACTION_DONE;
+
+public class OrderConfirmActivity extends TestpressFragmentActivity {
+
+    @Inject TestpressServiceProvider serviceProvider;
+    @Inject protected LogoutService logoutService;
+    @InjectView(id.address) EditText address;
+    @InjectView(id.zip) EditText zip;
+    @InjectView(id.landmark) EditText landmark;
+    @InjectView(id.phone) EditText phone;
+    @InjectView(id.fill_all_details) TextView fillAllDetailsText;
+    @InjectView(id.continueButton) Button continueButton;
+    @InjectView(id.pb_loading) ProgressBar progressBar;
+    @InjectView(id.shippingDetails) RelativeLayout shippingDetails;
+    final TextWatcher watcher = validationTextWatcher();
+    PaymentParams paymentParams;
+    PayuConfig payuConfig;
+    ProductDetails productDetails;
+    Order order;
+    OrderItem orderItem = new OrderItem();
+
+    @Override
+    public void onCreate(Bundle bundle) {
+        super.onCreate(bundle);
+        Injector.inject(this);
+        setContentView(R.layout.shipping_address);
+        ButterKnife.inject(this);
+        progressBar.setVisibility(View.VISIBLE);
+        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
+        shippingDetails.setVisibility(View.GONE);
+        progressBar.getIndeterminateDrawable().setColorFilter(getResources().getColor(R.color.primary), PorterDuff.Mode.SRC_IN);
+        productDetails = getIntent().getParcelableExtra("productDetails");
+        orderItem.setProduct(Constants.Http.URL_BASE + "/api/v2/products/" + productDetails.getSlug() + "/"); //now using v2 instead v2.1
```

Comment:
Please do not hardcode urls like these. Put it inside constants so that if we want to upgrade the api, we can do it in a single file.